cpp/Platform.Hashing/Platform.Hashing.Hash.h

```diff
@@ -3,34 +3,59 @@
 #ifndef PLATFORM_HASHING_HASH
 #define PLATFORM_HASHING_HASH
 
+#include <functional>
+
+#include "Platform.Hashing.Expand.h"
+
 namespace Platform::Hashing
 {
+
+
     template <typename T, typename = std::void_t<>>
-    struct is_std_hashable : std::false_type { };
+    struct is_std_hashable_t : std::false_type { };
+
+    template <typename T>
+    struct is_std_hashable_t<T, std::void_t<decltype(std::declval<std::hash<T>>()(std::declval<T>()))>> : std::true_type { };
+
+#ifndef __cpp_lib_concepts
+    template <typename T>
+    inline constexpr bool is_std_hashable = is_std_hashable_t<T>::value;
 
     template <typename T>
-    struct is_std_hashable<T, std::void_t<decltype(std::declval<std::hash<T>>()(std::declval<T>()))>> : std::true_type { };
+    inline constexpr bool not_std_hashable = !is_std_hashable_t<T>::value;
+#else
+
+    template <typename T>
+    concept is_std_hashable = requires(T object) {std::hash<T>{}(object);};
```

Comment:
Maybe the concepts should be called differently? It's not very cool when, depending on the standard, different things can be called by the same name, It confuses and complicates the readability of the code. I suggest this alternative:
```cpp
/// type
template <typename T, typename = std::void_t<>>
struct is_std_hashable : std::false_type { };

/// value
template <typename T>
inline constexpr bool is_std_hashable_v = is_std_hashable<T>::value;

/// concept
template <typename T>
concept std_hashable = requires(T object) {std::hash<T>{}(object);};
```
It looks more like standard [type traits](https://en.cppreference.com/w/cpp/types#Type_traits) and [concepts](https://en.cppreference.com/w/cpp/concepts)

Reply:
How then to ensure independence from the C++ version?

Reply:
We can just support the latest one at the moment.

Reply:
But then we can just leave the concepts and not think about it

Reply:
Concepts are more about modifying types in templates like ```template <std_hashable T>```, so I see two options:
1) It is possible inside the library in such necessary places to replace them with
```cpp
template <
#ifdef __cpp_concepts
std_hashable T
#else
typename T, typename = std::enable_if_t<is_std_hashable_v<T>
>
```
2) Just assume that lib without concepts doesn't work

Reply:
This is GNU syntax 🤣